identity-tools/developer_identity/create-dev-identity.sh

```diff
@@ -0,0 +1,256 @@
+
+export DEVID_CLI_DIR=$(cd $(dirname $0) && pwd)
+. "$DEVID_CLI_DIR/common.sh"
+
+cli_help() {
+  cli_name=${0##*/}
+  echo "
+  $cli_name - create pelion-edge developer identity
+  Version: $(cat $DEVID_CLI_DIR/VERSION)
+  Usage: $cli_name [options]
+
+  Options:
+
+    -h                            output usage information
+    -v                            verbose logging
+    -V                            output the version number
+    -d                            generate identity using default values
+    -m <lab_instance>             one of the following mbed-cloud lab instance (default: 'mbedcloud') -
+                                  [ mds-integration-lab OR mbedcloudintegration,
+                                    mds-systemtest OR mbedcloudstaging,
+                                    mbedcloud OR us-east-1.mbedcloud,
+                                    ap-northeast-1.mbedcloud ]
+    -c <lwm2m_coap_url>           lwm2m coap server address (default: 'coaps://lwm2m.us-east-1.mbedcloud.com')
+    -g <gw_server_url>            gateway services api address (default: 'https://gateways.us-east-1.mbedcloud.com')
+    -s <api_server_url>           api server address (default: 'https://api.us-east-1.mbedcloud.com')
+    -k <k8s_server_url>.          edge kubernetes server address (default: 'https://edge-k8s.us-east-1.mbedcloud.com')
+    -p <serial_number_prefix>     serial number prefix (default: 'DEV0')
+    -n <account_id>               account identifier (mandatory)
+    -o <output_directory>         output directory of identity.json (default: './')
+    -i <device_id>                edge-core's internal-id (mandatory)
+    -w <hw_type>                  hardware version of the gateway, refer configurations section in
+                                  $DEVID_CLI_DIR/radioProfile.template.json#L228 (default: 'rpi3bplus')
+    -r <radio_config>             radio configuration of the gateway, refer configurations section in
+                                  $DEVID_CLI_DIR/radioProfile.template.json#L228 (default: '00')
+    -l <led_config>               status led configuration of the gateway (default: '01')
+
+  Examples:
+
+    Typical usage, use the default values to generate an identity file. This will create random uuids for deviceID and accountID.
+
+        $cli_name -d
+
+    Generate a default identity file but with integration lab cloud addresses
+
+        $cli_name -d -m mbedcloudintegration
+
+    Generate a default identity file but relace the hardware version and radio configuration to specified value
+
+        $cli_name -d -w delledge3000 -r 01
+"
+}
+
+[ ! -n "$1" ] && cli_help && exit 1
+[ "$1" == "-v" ] && [ ! -n "$2" ] && cli_help && exit 1
+
+OPTIND=1
+
+while getopts 'hvVdm:c:g:s:k:p:n:o:i:w:r:l:' opt; do
+    case "$opt" in
+        h|-help)
+            cli_help
+            exit 0
+            ;;
+        v)
+            VERBOSE=1
+            ;;
+        V)
+            echo "$(cat $DEVID_CLI_DIR/VERSION)"
```

Comment:
Is there a reason that `VERSION` is a discrete file instead of baked into the script?

Reply:
Just so that it can be easily found and not get buried into the script.